src/notebook/notebookKernel.ts

```diff
@@ -30,182 +28,144 @@ export class JuliaKernel implements vscode.NotebookKernel {
+	}
+
+	async executeCell(cell: vscode.NotebookCell): Promise<void> {
+	    const task = vscode.notebook.createNotebookCellExecutionTask(cell.notebook.uri, cell.index, this.id)
+	    task.clearOutput()
+	    await this.start()
+	    const startTime = Date.now()
+	    task.start({startTime})
+	    const executionOrder = ++this._current_request_id
+	    task.executionOrder = executionOrder
+	    this.executionRequests.set(executionOrder, { startTime, task })
+	    this._msgConnection.sendNotification(notifyTypeRunCell, { current_request_id: this._current_request_id, code: cell.document.getText() })
+	}
+
+	public async start() {
+	    if (!this._terminal) {
+	        const connectedPromise = new Subject()
+	        const serverListeningPromise = new Subject()
+
+	        const pn = generatePipeName(uuid(), 'vscjl-nbk')
+
+	        const server = net.createServer(socket => {
+	            this._msgConnection = createMessageConnection(
+	                new StreamMessageReader(socket),
+	                new StreamMessageWriter(socket)
+	            )
+
+	            this._msgConnection.onNotification(notifyTypeRunCellSucceeded, ({ request_id }) => {
+	                const runEndTime = Date.now()
+	                const { task, startTime } = this.executionRequests.get(request_id)
+	                task.end({ success: true, duration: runEndTime - startTime })
+	            })
+
+	            this._msgConnection.onNotification(notifyTypeRunCellFailed, ({ request_id, output }) => {
+	                const runEndTime = Date.now()
+	                const { task, startTime } = this.executionRequests.get(request_id)
+	                task.appendOutput([new vscode.NotebookCellOutput([new vscode.NotebookCellOutputItem('application/x.notebook.error-traceback', output)])])
+	                task.end({ success: false, duration: runEndTime - startTime })
+	            })
+
+	            this._msgConnection.onNotification(notifyTypeDisplay, ({ mimetype, current_request_id, data }) => {
+	                const executionRequest = this.executionRequests.get(current_request_id)
+
+	                if (executionRequest) {
+	                    executionRequest.task.replaceOutput([new vscode.NotebookCellOutput([new vscode.NotebookCellOutputItem(mimetype, data)])])
+	                }
+	            })
+
+	            this._msgConnection.onNotification(notifyTypeStreamoutput, ({ name, current_request_id, data }) => {
+	                if (name === 'stdout') {
+	                    const executionRequest = this.executionRequests.get(current_request_id)
+	                    if (executionRequest) {
+	                        executionRequest.task.replaceOutput([new vscode.NotebookCellOutput([new vscode.NotebookCellOutputItem('application/x.notebook.stdout', data)])])
+	                    }
+	                }
+	                else if (name === 'stderr') {
+	                    const executionRequest = this.executionRequests.get(current_request_id)
```

Comment:
Same as earlier, depends on what stream packages write to when generating progress bars.
I've found plenty of pacakges using stdout & stderr in python.